src/mujoco_system_interface.cpp

```diff
@@ -178,71 +188,125 @@ const char* Diverged(int disableflags, const mjData* d)
   return nullptr;
 }
 
-mjModel* LoadModel(const char* file, mj::Simulate& sim)
+mjModel* LoadModel(const char* file, mj::Simulate& sim, rclcpp::Node::SharedPtr node)
 {
+  mjModel* mnew = 0;
+
+  // Try to get the mujoco model from URDF
```

Comment:
Sorry I'm a big fan of explanatory comments for complex functions. Can you just add something like the following to explain how this function works?

```suggestion
  // Try to get the mujoco model from URDF.
  // If it is not available, create a subscription and listen for the model on a topic.
```